common/flatpak-run.c

```diff
@@ -2227,16 +2227,36 @@ flatpak_run_setup_base_argv (FlatpakBwrap   *bwrap,
                           "--perms", "0700", "--dir", run_dir,
                           "--setenv", "XDG_RUNTIME_DIR", run_dir,
                           "--symlink", "../run", "/var/run",
-                          "--ro-bind", "/sys/block", "/sys/block",
-                          "--ro-bind", "/sys/bus", "/sys/bus",
-                          "--ro-bind", "/sys/class", "/sys/class",
-                          "--ro-bind", "/sys/dev", "/sys/dev",
-                          "--ro-bind", "/sys/devices", "/sys/devices",
                           "--ro-bind-try", "/proc/self/ns/user", "/run/.userns",
                           /* glib uses this like /etc/timezone */
                           "--symlink", "/etc/timezone", "/var/db/zoneinfo",
                           NULL);
 
+  if (access ("/sys/block", R_OK|X_OK) == 0)
+    flatpak_bwrap_add_args (bwrap,
+                            "--ro-bind", "/sys/block", "/sys/block",
+                            NULL);
```

Comment:
If Flatpak maintainers and app maintainers are going to have any hope of triaging or diagnosing app issues that are triggered by being unable to enumerate devices in `/sys`, this is going to need to log a message, at least when using `flatpak -v run`:

```suggestion
                            NULL);
    else
      g_info ("Not sharing %s with sandbox: %s", "/sys/block", g_strerror (errno));
```

(But combined with my other suggestion, you'd need to use `dir` rather than `"/sys/block"`.)

Reply:
`g_info()` is not logged by default, but we do log it under `flatpak -v` or higher.

(For comparison, `g_debug()` is only logged under `flatpak -v -v` or higher: that's probably hiding away this information too much. Meanwhile, higher-severity messages like `g_warning()` and `g_message()` are logged by default: probably too noisy in this case.)